internal/installation/paths.go

```diff
@@ -62,13 +61,31 @@ func BinPathFromInstallPath(installPath string) (string, error) {
 	return filepath.Join(installPath, BinDirName), nil
 }
 
-func InstalledOnPath(installPath string) (bool, string, error) {
-	binPath, err := BinPathFromInstallPath(installPath)
+func InstalledOnPath(installRoot string) (bool, string, error) {
+	if !fileutils.DirExists(installRoot) {
+		return false, "", nil
+	}
+
+	path := appinfo.StateApp(installRoot).Exec()
+	if fileutils.TargetExists(path) {
+		return true, filepath.Dir(path), nil
```

Comment:
Right now I'm using the `InstallDirMarker` file as a fallback because older versions of the state tool will not have that file when updating.

Reply:
Please move any sort of logic dealing with old installation formats to the updater. The point of this story is that we can get rid of all this type of fallback / legacy logic and assume a pristine installation structure.

This could very well mean moving the entire `InstalledOnPath()` function into the updater/installer because that's the only place it's used. Point here is that the installer is the ONLY part of the code vulnerable to legacy or corrupted install dir formats.